tests/test_uri.py

```diff
@@ -343,6 +395,40 @@ def testTransfer(self):
 
         dest.transfer_from(src, transfer="copy", overwrite=True)
 
+    def testWalk(self):
+        """Test that we can list an S3 bucket"""
+        # Files we want to create
+        expected = ("a/x.txt", "a/y.txt", "a/z.json", "a/b/w.txt", "a/b/c/d/v.json")
```

Comment:
I think you should have at least one test that extends over more than one page, which unfortunately means a lot of files, given the default max of 1000 keys.